### DISCUSSION TITLE
Error 401 while try to create Oauth2 client or Personnal Token

### DISCUSSION THREAD

**Discussion by Gsyltc** [firefly-iii/firefly-iii#6775] [Firefly III]
I think I've seen several posts on the subject, but they don't seem to be identical to my problem.

I am trying to create an Oauth2 client for an application (n8n).
Here is my configuration:

Installtion : Docker
Reverse Proxy : Traefik
Authentication : Authelia
     I have bypassed
        - "^/api([/?].*)?$"
        - "^/manifest.webmanifest([/?].*)?$"

When I try to create a clien, I get a 401 error in the firefly iii logs.
`10.0.6.95 - - [30/Dec/2022:13:33:08 -0500] "POST /oauth/clients HTTP/1.1" 401 1874 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36"`

There is no exception  

I checked my authelia logs. Everything is correct, it's not authelia that seems to block authentication.

Did I forget to do something? 
Thanks

**Comment by JC5** [comment]
Try excluding `/oauth/clients`.

**Comment by Gsyltc** [comment]
Hello, 

I already tried to put "^/oauth([/?].*)?$"  
But it doesn't change anything.

I think that the problem could come from somewhere else than Authelia. Here are my tests:

Launching a new instance with the same parameters (env var, docker, authelia, etc.) with a web authentication.
I manage to create the personal token and an Oauth2 client, I receive the confirmation mail and n8n can connect to the API. all is good

I restart the same instance with a remote user authentication, the new account is not able to create the tokens or the clients.
I get the error 401. 

It doesn't seem to come from the environment configuration but from an application level issue.
Have you encountered this problem before?

**Reply by JC5** [reply]
No, I haven't, sorry. I'll see what I can find myself.

**Comment by Gsyltc** [comment]
Ok; I will make some test tomorrow in order to detect something to help

**Reply by JC5** [reply]
I think I found the issue, and I've pushed a fix to the `develop` tag.

**Comment by Gsyltc** [comment]
Ok, I'll try it tomorrow morning :)
Thanks and Good night

**Comment by Gsyltc** [comment]
Hi,

I tried the develop branch (with the new docker image "develop").
Unfortunatly, nothing change :

I saw a thing. 401 errors does appaers when I click on "profile". I have these 401 errors
```
10.0.6.95 - - [01/Jan/2023:01:27:34 -0500] "GET /oauth/tokens HTTP/1.1" 401 1874 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36"
10.0.6.95 - - [01/Jan/2023:01:27:34 -0500] "GET /oauth/clients HTTP/1.1" 401 1874 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36"
10.0.6.95 - - [01/Jan/2023:01:27:34 -0500] "GET /oauth/scopes HTTP/1.1" 401 1874 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36"
10.0.6.95 - - [01/Jan/2023:01:27:34 -0500] "GET /oauth/personal-access-tokens HTTP/1.1" 401 1874 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36"
```

All Oauth path are bypassed in authelia.

When I click on OAuth page, I have these log 

```
[2023-01-01T01:29:58.591411-05:00] production.INFO: AUDIT: Show login form (1.1). (XXX.XXX.XXX.XXX -> GET:https://firefly.mydomainname/login) [] []
```

and when I click on Create token (for exemple), I have this log 

```
10.0.6.95 - - [01/Jan/2023:01:32:02 -0500] "POST /oauth/personal-access-tokens HTTP/1.1" 401 1874 "-" "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/108.0.0.0 Safari/537.36"
```

So 401 error appears multiple times.

I hope this can help you.

Thanks

Have a nice day

**Reply by JC5** [reply]
I see what you mean, I'll check it out!

**Comment by JC5** [comment]
Tracking this in https://github.com/firefly-iii/firefly-iii/issues/6788

**Comment by Gsyltc** [comment]
Hi @JC5 Good news, It's works
I can Create OAuth client ands token.

Thanks
